lib/streamlit/elements/select_slider.py

```diff
@@ -0,0 +1,114 @@
+from streamlit.proto.Slider_pb2 import Slider as SliderProto
+from streamlit.errors import StreamlitAPIException
+from streamlit.type_util import ensure_iterable
+from .utils import _get_widget_ui_value
+
+
+class SelectSliderMixin:
+    def select_slider(
+        dg, label, options=[], value=None, format_func=str, key=None,
+    ):
+        """
+        Display a slider widget to select items from a list.
+
+        This also allows you to render a range slider by passing a two-element
+        tuple or list as the `value`.
+
+        The difference between `st.select_slider` and `st.slider` is that
+        `select_slider` accepts any datatype and takes a list as input, while
+        `slider` only accepts numerical or date/time data and takes a range as
+        input.
+
+        Parameters
+        ----------
+        label : str or None
+            A short label explaining to the user what this slider is for.
+        options : list, tuple, numpy.ndarray, pandas.Series, or pandas.DataFrame
+            Labels for the slider options. All options will be cast to str
+            internally by default. For pandas.DataFrame, the first column is
+            selected.
+        value : a supported type or a tuple/list of supported types or None
+            The value of the slider when it first renders. If a tuple/list
+            of two values is passed here, then a range slider with those lower
+            and upper bounds is rendered. For example, if set to `(1, 10)` the
```

Comment:
do we need to be more careful here in our design, since other Python intervals are half-open? range(1,10) in Python is 1..9

Reply:
I don't think this should be a problem. I think the word `range` is overloaded in this case, but sliders can be used to represent ranges where as python ranges are a different concept.